app/octopus_to_influxdb.py

```diff
@@ -48,6 +51,10 @@ def __init__(self, config_file=None):
             config.get('octopus', 'cache_dir', fallback='/tmp/octopus_api_cache') if config.getboolean('octopus', 'enable_cache', fallback=False) else None
         )
 
+        self._octopus_ql_api = OctopusEnergyApiClient(
```

Comment:
Can you only initialise this if it's required, i.e. 
`if self._intelligent_tariff_meter is not None:`

This will require moving the `_intelligent_tariff_meter` initialisation further up.